Form validation: Normalize required fieldsets

EricDunsworth opened the pull request:
#9091 changed some of the form validation demo page's required fieldsets to use ``data-rule-required="true"`` attributes on their first fields (instead of ``required="required"``). But it left out the stacked/horizontal fieldset examples and the server demo page.

This changes the leftovers to use ``data-rule-required="true"`` for consistency. Also adds its alert nearby.

Related changes:
* Made the alert's wording more specific... now specifically mentions required fieldsets (it was previously ambiguous - which might've caused confusion when the ``aria-hidden="true"`` alert was nearby)
* Changed the alert into an include (similar to what's already being done with the ``aria-hidden="true"`` alert)
* Replaced the ``aria-hidden="true"`` alert with the ``data-rule-required="true"`` alert in #9866's no asterisk fieldset example (should've used the latter in the first place but overlooked it)

CC @thekodester @StdGit @delisma

thekodester commented:
Make sense!

StdGit commented:
@EricDunsworth Did you test https://wet-boew.github.io/wet-boew/demos/formvalid/formvalid-server-en.html ?

GormFrank commented:
@EricDunsworth Please rebase again 🥇  @duboisp are we good to god after?

EricDunsworth commented:
Just rebased it :).

duboisp commented:
We will need to describe the use case, what is the problematic to the user, how we can test, what is the current result, what is the expected result. What it is the context?  I can help you to write a design decision during our weekly code sprint.

Garneauma commented:
@EricDunsworth Is this PR still relevant? If yes, I will do a new review. Thanks!

EricDunsworth commented:
@Garneauma I'd consider it to still be relevant :).

What held it up last time was a request for a design decision... I think I was against it at the time since since the original goal of this PR was just to cover some spots that were missed by #9091. I didn't feel like it made sense to merge that one in "as-is" and necessitate a full-blown design decision for this one :P.

**PS:**
Just rebased, added a few updates and force-pushed:
* Resolved conflicts
* Made the alert's wording more specific... now specifically mentions required fieldsets (it was previously ambiguous - which might've caused confusion when the [``aria-hidden="true"`` alert](https://github.com/wet-boew/wet-boew/blob/master/site/includes/alertariahidden.hbs) was nearby)
* Changed the alert into an include (similar to what's already being done with the ``aria-hidden="true"`` alert)
* Added ``data-rule-required="true"`` and the alert to several more fieldset demos (they probably didn't exist when this PR was first created)
* Added the alert to the [merge server-client errors demo page](https://wet-boew.github.io/v4.0-ci/demos/formvalid/formvalid-server-en.html)
* Replaced the ``aria-hidden="true"`` alert with the ``data-rule-required="true"`` alert in #9866's no asterisk fieldset example (should've used the latter in the first place but overlooked it)
* Updated modified dates

Garneauma commented:
@EricDunsworth Approved upon rebasing. As far as I can see, this is just content change that was already done on other examples that were missed, correct?

EricDunsworth commented:
Just rebased :).

All the conflicts seemed to be related to #9977's trailing slash removals from void (self-closing) element tags. Also updated modified dates.

> As far as I can see, this is just content change that was already done on other examples that were missed, correct?

@Garneauma That's correct. I noticed that #9091 hadn't applied its change to all required fieldsets and wanted to address the "leftovers" with this PR (to make it consistent everywhere).

FWIW #9091's creator (@ghost - aka @kodecount / @thekodester) was on board with it. The ["merged" server-client errors template](https://wet-boew.github.io/v4.0-ci/demos/formvalid/formvalid-server-en.html)'s creator (@StdGit) was also ok with it.

From my recollection, I think the reason this PR wasn't merged back in 2021 was two-fold:
* IIRC @duboisp wanted to show a "mix" of different ways of handling required fieldsets in the examples. I think he preferred the status quo at the time.
* @duboisp requested a [design decision](https://wet-boew.github.io/wet-boew-documentation/decision/index-en.html) in https://github.com/wet-boew/wet-boew/pull/9161#issuecomment-983983822 - which I never got around to writing :(.
  * **Note:** I think I also questioned the need for the design decision since my PR wasn't _introducing_ ``data-rule-required="true"`` nor the alert into the examples (they had already been introduced via #9091 - which I had no involvement with). So my line of thinking was "why should I have to justify someone else's already-merged change :S?".

Garneauma commented:
@Yves-Denis For your review. Only content change.

Garneauma commented:
@EricDunsworth You'll need to rebase again because of your other content PR. Sorry.

EricDunsworth committed:
Form validation: Normalize required fieldsets

#9091 changed some of the form validation demo page's required fieldsets to use data-rule-required="true" attributes on their first fields (instead of required="required"). But it left out the stacked/horizontal fieldset examples and the server demo page.

This changes the leftovers to use data-rule-required="true" for consistency. Also adds its alert nearby.

Related changes:
* Made the alert's wording more specific... now specifically mentions required fieldsets (it was previously ambiguous - which might've caused confusion when the aria-hidden="true" alert was nearby)
* Changed the alert into an include (similar to what's already being done with the aria-hidden="true" alert)
* Replaced the aria-hidden="true" alert with the data-rule-required="true" alert in #9866's no asterisk fieldset example (should've used the latter in the first place but overlooked it)

EricDunsworth commented:
@Garneauma Just rebased, resolved the conflicts, updated modified dates and force-pushed :).

Yves-Denis reviewed (approved):
Security Review Completed - Passed, no issues identified
